pkg/cmd/pulumi/schema.go

```diff
@@ -31,5 +39,43 @@ package schemas for errors.`,
 	}
 
 	cmd.AddCommand(newSchemaCheckCommand())
+	cmd.AddCommand(newSchemaGenerateSDKCommand())
 	return cmd
 }
+
+func loadSchemaSpec(path, kind string) (schema.PackageSpec, error) {
+	// Read from stdin or a specified file
+	reader := os.Stdin
+	if path != "-" {
+		f, err := os.Open(path)
+		if err != nil {
+			return schema.PackageSpec{}, fmt.Errorf("could not open %v: %w", path, err)
+		}
+		defer f.Close()
+
+		reader = f
+	}
+	schemaBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return schema.PackageSpec{}, fmt.Errorf("failed to read schema: %w", err)
+	}
+
+	var pkgSpec schema.PackageSpec
+	if kind == "" {
+		if ext := filepath.Ext(path); ext == ".yaml" || ext == ".yml" {
+			kind = "yaml"
+		} else {
```

Comment:
I'm hesitant to assume `json` if not `yaml`. 
```suggestion
              } else if ext == ".json" {
```
Feel free to ignore this though.